clients/kafka-common/src/main/java/org/eclipse/hono/client/kafka/KafkaProducerFactory.java

```diff
@@ -67,6 +67,25 @@
      */
     KafkaProducer<K, V> getOrCreateProducer(String producerName, Map<String, String> config);
 
+    /**
+     * Gets a producer for sending data to Kafka.
+     * <p>
+     * The producer returned may be either newly created or it may be an existing producer for the given producer name.
+     * The config parameter might be ignored if an existing producer is returned.
+     * <p>
+     * Do not hold references to the returned producer between send operations, because the producer might be closed by
+     * the factory. Instead, always get an instance by invoking this method.
+     * <p>
+     * When the producer is no longer needed, it should be closed to release the resources (like connections). NB: Do
+     * not close the returned producer directly. Instead, call {@link #closeProducer(String)} to let the factory close
+     * the producer and update its internal state.
+     *
+     * @param producerName The name to identify the producer.
+     * @param config The Kafka configuration with which the producer is to be created.
+     * @return an existing or new producer.
+     */
+    KafkaProducer<K, V> getOrCreateProducer(String producerName, KafkaProducerConfigProperties config);
```

Comment:
FMPOV the other variant with the map parameter should be entirely be removed.

Reply:
done